diag_manager/fms_diag_object.F90

```diff
@@ -553,9 +564,14 @@ logical function fms_diag_accept_data (this, diag_field_id, field_data, time, is
 !!TODO: Loop through fields and do averages/math functions
     do i = 1, size(this%FMS_diag_fields(diag_field_id)%buffer_ids)
       buffer_id = this%FMS_diag_fields(diag_field_id)%buffer_ids(i)
+      file_id = this%FMS_diag_fields(diag_field_id)%file_ids(i)
```

Comment:
Everything you add here needs to also go in `fms_diag_send_complete` because in the `main_if: if (buffer_the_data) then` statement when `buffer_the_data` is `.true.`, all of the math reductions will be done in `fms_diag_send_complete`.  I think you should put as much as possible contained in this `else` block into a function/subroutine to avoid duplicate code.

Reply:
You could even put this into a new module or object to handle just the math functions.

Reply:
You are right. I will move the whole block to a new subroutine.